app/src/main/java/com/example/cattlelog/MainActivity.kt

```diff
@@ -86,15 +106,59 @@ class MainActivity : AppCompatActivity() {
             cattleList?.let { adapter.setCattleList(it) }
         })
 
+    }
 
-//        recyclerView = findViewById<RecyclerView>(R.id.herdList).apply {
-//            setHasFixedSize(true)
-//            layoutManager = viewManager
-//            adapter = viewAdapter
-//        }
+    private fun checkDatabaseVersion() {
+        var newestDBVersion = ""
+        Log.d("dropboxAPI", "entered check file function.")
+        Log.d("dropboxAPI", resources.getString(R.string.CattleLog_DropboxAPIKey))
+
+
+        val url = "https://api.dropboxapi.com/2/files/get_metadata"
+        val params = HashMap<String,String>()
+        params["path"] = "/cattlelog_database.db"
+        val jsonObject = JSONObject(params)
+
+        // Some help from https://android--code.blogspot.com/2019/02/android-kotlin-volley-post-request-with.html
+        // Volley post request with parameters
+        val request = object: JsonObjectRequest(Method.POST,url,jsonObject,
+            Response.Listener { response ->
+                // Process the json
+                try {
+                    Log.d("dropboxAPI", "Response: $response")
+                    newestDBVersion = response["server_modified"].toString()
+                    Log.d("dropboxAPI", "Last Modification of CattleLog DB: $newestDBVersion")
+                }catch (e:Exception){
+                    Log.d("dropboxAPI", "Exception: $e")
+                }
+
+            }, Response.ErrorListener{
+                // Error in request
+                Log.d("dropboxAPI", "Volley error: $it")
+            }){
 
+            override fun getHeaders(): HashMap<String, String> {
+                val headers = HashMap<String, String>()
+                headers["Authorization"] = "Bearer " + resources.getString(R.string.CattleLog_DropboxAPIKey)
+                return headers
+            }
+
+        }
+
+        // Volley request policy, only one time request to avoid duplicate transaction
+        request.retryPolicy = DefaultRetryPolicy(
+            DefaultRetryPolicy.DEFAULT_TIMEOUT_MS,
+            // 0 means no retry
+            0, // DefaultRetryPolicy.DEFAULT_MAX_RETRIES = 2
+            1f // DefaultRetryPolicy.DEFAULT_BACKOFF_MULT
```

Comment:
What are the comments for next to the `0` and `1f`?

Reply:
Was from copied boilerplate.  Probably unnecessary since android studio annotates the parameters anyway.

Reply:
+1